<p style="text-align:center">
    <a href="https://skills.network/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkPY0220ENSkillsNetwork900-2022-01-01" target="_blank">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/assets/logos/SN_web_lightmode.png" width="200" alt="Skills Network Logo">
    </a>
</p>


<h1>Extracting and Visualizing Stock Data</h1>
<h2>Description</h2>


Extracting essential data from a dataset and displaying it is a necessary part of data science; therefore individuals can make correct decisions based on the data. In this assignment, you will extract some stock data, you will then display this data in a graph.


<h2>Table of Contents</h2>
<div class="alert alert-block alert-info" style="margin-top: 20px">
    <ul>
        <li>Define a Function that Makes a Graph</li>
        <li>Question 1: Use yfinance to Extract Stock Data</li>
        <li>Question 2: Use Webscraping to Extract Tesla Revenue Data</li>
        <li>Question 3: Use yfinance to Extract Stock Data</li>
        <li>Question 4: Use Webscraping to Extract GME Revenue Data</li>
        <li>Question 5: Plot Tesla Stock Graph</li>
        <li>Question 6: Plot GameStop Stock Graph</li>
    </ul>
<p>
    Estimated Time Needed: <strong>30 min</strong></p>
</div>

<hr>


***Note***:- If you are working Locally using anaconda, please uncomment the following code and execute it.


In [1]:
!pip install yfinance==0.2.38
!pip install pandas==2.2.2
!pip install nbformat

     ---------------------------------------- 0.0/3.0 MB ? eta -:--:--
     ---------------------------------------- 0.0/3.0 MB ? eta -:--:--
      --------------------------------------- 0.1/3.0 MB 825.8 kB/s eta 0:00:04
     - -------------------------------------- 0.1/3.0 MB 1.0 MB/s eta 0:00:03
     -- ------------------------------------- 0.2/3.0 MB 919.0 kB/s eta 0:00:04
     -- ------------------------------------- 0.2/3.0 MB 1.0 MB/s eta 0:00:03
     --- ------------------------------------ 0.3/3.0 MB 1.2 MB/s eta 0:00:03
     ----- ---------------------------------- 0.4/3.0 MB 1.3 MB/s eta 0:00:02
     ------ --------------------------------- 0.5/3.0 MB 1.4 MB/s eta 0:00:02
     -------- ------------------------------- 0.7/3.0 MB 1.6 MB/s eta 0:00:02
     ---------- ----------------------------- 0.8/3.0 MB 1.7 MB/s eta 0:00:02
     ----------- ---------------------------- 0.9/3.0 MB 1.7 MB/s eta 0:00:02
     ------------ --------------------------- 1.0/3.0 MB 1.8 MB/s eta 0:00

In [2]:
!pip install yfinance==0.1.67
!mamba install bs4==4.10.0 -y
!pip install nbformat==4.2.0

  Using cached yfinance-0.1.67-py2.py3-none-any.whl.metadata (10 kB)
Using cached yfinance-0.1.67-py2.py3-none-any.whl (25 kB)


'mamba' is not recognized as an internal or external command,
operable program or batch file.


  Using cached nbformat-4.2.0-py2.py3-none-any.whl.metadata (1.1 kB)
Using cached nbformat-4.2.0-py2.py3-none-any.whl (153 kB)
  Attempting uninstall: nbformat
    Found existing installation: nbformat 5.9.2
    Uninstalling nbformat-5.9.2:
      Successfully uninstalled nbformat-5.9.2


ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
conda-repo-cli 1.0.75 requires requests_mock, which is not installed.
anaconda-client 1.12.3 requires nbformat>=4.4.0, but you have nbformat 4.2.0 which is incompatible.
conda-repo-cli 1.0.75 requires clyent==1.2.1, but you have clyent 1.2.2 which is incompatible.
conda-repo-cli 1.0.75 requires nbformat==5.9.2, but you have nbformat 4.2.0 which is incompatible.
conda-repo-cli 1.0.75 requires python-dateutil==2.8.2, but you have python-dateutil 2.9.0.post0 which is incompatible.
jupyter-server 2.10.0 requires nbformat>=5.3.0, but you have nbformat 4.2.0 which is incompatible.
nbclient 0.8.0 requires nbformat>=5.1, but you have nbformat 4.2.0 which is incompatible.
nbconvert 7.10.0 requires nbformat>=5.7, but you have nbformat 4.2.0 which is incompatible.
spyder 5.4.3 requires jedi<0.19.0,>=0.17.2, but you have jedi

In [13]:
import yfinance as yf
import pandas as pd
import requests

from datetime import datetime ,timedelta

from bs4 import BeautifulSoup
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In Python, you can ignore warnings using the warnings module. You can use the filterwarnings function to filter or ignore specific warning messages or categories.


In [4]:
import warnings
# Ignore all warnings
warnings.filterwarnings("ignore", category=FutureWarning)

## Define Graphing Function


In this section, we define the function `make_graph`. **You don't have to know how the function works, you should only care about the inputs. It takes a dataframe with stock data (dataframe must contain Date and Close columns), a dataframe with revenue data (dataframe must contain Date and Revenue columns), and the name of the stock.**


In [5]:
def make_graph(stock_data, revenue_data, stock):
    fig = make_subplots(rows=2, cols=1, shared_xaxes=True, subplot_titles=("Historical Share Price", "Historical Revenue"), vertical_spacing = .3)
    stock_data_specific = stock_data[stock_data.Date <= '2021--06-14']
    revenue_data_specific = revenue_data[revenue_data.Date <= '2021-04-30']
    fig.add_trace(go.Scatter(x=pd.to_datetime(stock_data_specific.Date, infer_datetime_format=True), y=stock_data_specific.Close.astype("float"), name="Share Price"), row=1, col=1)
    fig.add_trace(go.Scatter(x=pd.to_datetime(revenue_data_specific.Date, infer_datetime_format=True), y=revenue_data_specific.Revenue.astype("float"), name="Revenue"), row=2, col=1)
    fig.update_xaxes(title_text="Date", row=1, col=1)
    fig.update_xaxes(title_text="Date", row=2, col=1)
    fig.update_yaxes(title_text="Price ($US)", row=1, col=1)
    fig.update_yaxes(title_text="Revenue ($US Millions)", row=2, col=1)
    fig.update_layout(showlegend=False,
    height=900,
    title=stock,
    xaxis_rangeslider_visible=True)
    fig.show()

Use the make_graph function that we’ve already defined. You’ll need to invoke it in questions 5 and 6 to display the graphs and create the dashboard. 
> **Note: You don’t need to redefine the function for plotting graphs anywhere else in this notebook; just use the existing function.**


## Question 1: Use yfinance to Extract Stock Data


Using the `Ticker` function enter the ticker symbol of the stock we want to extract data on to create a ticker object. The stock is Tesla and its ticker symbol is `TSLA`.


In [17]:
end_date = datetime.today()
end_date

datetime.datetime(2025, 2, 6, 15, 19, 13, 731172)

In [27]:
start_date = end_date - timedelta(days=2*360) 
start_date

datetime.datetime(2023, 2, 17, 15, 19, 13, 731172)

In [28]:
yf.download('TSLA', start=start_date, end=end_date)

[*********************100%%**********************]  1 of 1 completed


Open        High         Low       Close   Adj Close  \
Date                                                                     
2023-02-17  199.990005  208.440002  197.500000  208.309998  208.309998   
2023-02-21  204.990005  209.710007  197.220001  197.369995  197.369995   
2023-02-22  197.929993  201.990005  191.779999  200.860001  200.860001   
2023-02-23  203.910004  205.139999  196.330002  202.070007  202.070007   
2023-02-24  196.330002  197.669998  192.800003  196.880005  196.880005   
...                ...         ...         ...         ...         ...   
2025-01-30  410.779999  412.500000  384.410004  400.279999  400.279999   
2025-01-31  401.529999  419.989990  401.339996  404.600006  404.600006   
2025-02-03  386.679993  389.170013  374.359985  383.679993  383.679993   
2025-02-04  382.630005  394.000000  381.399994  392.209991  392.209991   
2025-02-05  387.510010  388.390015  375.529999  378.170013  378.170013   

               Volume  
Date                   
2023-02-17  213738500  
2023-02-21  180018600  
2023-02-22  191828500  
2023-02-23  146360000  
2023-02-24  142228100  
...               ...  
2025-01-30   98092900  
2025-01-31   83568200  
2025-02-03   93732100  
2025-02-04   57072200  
2025-02-05   57223300  

[493 rows x 6 columns]

In [33]:
tsla = yf.Ticker("TSLA")

In [34]:
tsla.info

{'address1': '1 Tesla Road',
 'city': 'Austin',
 'state': 'TX',
 'zip': '78725',
 'country': 'United States',
 'phone': '512 516 8177',
 'website': 'https://www.tesla.com',
 'industry': 'Auto Manufacturers',
 'industryKey': 'auto-manufacturers',
 'industryDisp': 'Auto Manufacturers',
 'sector': 'Consumer Cyclical',
 'sectorKey': 'consumer-cyclical',
 'sectorDisp': 'Consumer Cyclical',
 'longBusinessSummary': 'Tesla, Inc. designs, develops, manufactures, leases, and sells electric vehicles, and energy generation and storage systems in the United States, China, and internationally. The company operates in two segments, Automotive, and Energy Generation and Storage. The Automotive segment offers electric vehicles, as well as sells automotive regulatory credits; and non-warranty after-sales vehicle, used vehicles, body shop and parts, supercharging, retail merchandise, and vehicle insurance services. This segment also provides sedans and sport utility vehicles through direct and used vehic

Using the ticker object and the function `history` extract stock information and save it in a dataframe named `tesla_data`. Set the `period` parameter to ` "max" ` so we get information for the maximum amount of time.


In [38]:
tesla_data = pd.DataFrame(tsla.history())
tesla_data

Open        High         Low       Close  \
Date                                                                        
2025-01-06 00:00:00-05:00  423.200012  426.429993  401.700012  411.049988   
2025-01-07 00:00:00-05:00  405.829987  414.329987  390.000000  394.359985   
2025-01-08 00:00:00-05:00  392.950012  402.500000  387.399994  394.940002   
2025-01-10 00:00:00-05:00  391.399994  399.279999  377.290009  394.739990   
2025-01-13 00:00:00-05:00  383.209991  403.790009  380.070007  403.309998   
2025-01-14 00:00:00-05:00  414.339996  422.640015  394.540009  396.359985   
2025-01-15 00:00:00-05:00  409.899994  429.799988  405.660004  428.220001   
2025-01-16 00:00:00-05:00  423.489990  424.000000  409.130005  413.820007   
2025-01-17 00:00:00-05:00  421.500000  439.739990  419.750000  426.500000   
2025-01-21 00:00:00-05:00  432.640015  433.200012  406.309998  424.070007   
2025-01-22 00:00:00-05:00  416.809998  428.000000  414.589996  415.109985   
2025-01-23 00:00:00-05:00  416.059998  420.730011  408.950012  412.380005   
2025-01-24 00:00:00-05:00  414.450012  418.880005  405.779999  406.579987   
2025-01-27 00:00:00-05:00  394.799988  406.690002  389.000000  397.149994   
2025-01-28 00:00:00-05:00  396.910004  400.589996  386.500000  398.089996   
2025-01-29 00:00:00-05:00  395.209991  398.589996  384.480011  389.100006   
2025-01-30 00:00:00-05:00  410.779999  412.500000  384.410004  400.279999   
2025-01-31 00:00:00-05:00  401.529999  419.989990  401.339996  404.600006   
2025-02-03 00:00:00-05:00  386.679993  389.170013  374.359985  383.679993   
2025-02-04 00:00:00-05:00  382.630005  394.000000  381.399994  392.209991   
2025-02-05 00:00:00-05:00  387.510010  388.390015  375.529999  378.170013   

                             Volume  Dividends  Stock Splits  
Date                                                          
2025-01-06 00:00:00-05:00  85516500        0.0           0.0  
2025-01-07 00:00:00-05:00  75699500        0.0           0.0  
2025-01-08 00:00:00-05:00  73038800        0.0           0.0  
2025-01-10 00:00:00-05:00  62287300        0.0           0.0  
2025-01-13 00:00:00-05:00  67580500        0.0           0.0  
2025-01-14 00:00:00-05:00  84565000        0.0           0.0  
2025-01-15 00:00:00-05:00  81375500        0.0           0.0  
2025-01-16 00:00:00-05:00  68335200        0.0           0.0  
2025-01-17 00:00:00-05:00  94991400        0.0           0.0  
2025-01-21 00:00:00-05:00  87320900        0.0           0.0  
2025-01-22 00:00:00-05:00  60963300        0.0           0.0  
2025-01-23 00:00:00-05:00  50690600        0.0           0.0  
2025-01-24 00:00:00-05:00  56427100        0.0           0.0  
2025-01-27 00:00:00-05:00  58125500        0.0           0.0  
2025-01-28 00:00:00-05:00  48910700        0.0           0.0  
2025-01-29 00:00:00-05:00  68033600        0.0           0.0  
2025-01-30 00:00:00-05:00  98092900        0.0           0.0  
2025-01-31 00:00:00-05:00  83568200        0.0           0.0  
2025-02-03 00:00:00-05:00  93732100        0.0           0.0  
2025-02-04 00:00:00-05:00  57072200        0.0           0.0  
2025-02-05 00:00:00-05:00  57223300        0.0           0.0

**Reset the index** using the `reset_index(inplace=True)` function on the tesla_data DataFrame and display the first five rows of the `tesla_data` dataframe using the `head` function. Take a screenshot of the results and code from the beginning of Question 1 to the results below.


In [39]:
tesla_data.reset_index(inplace=True)

In [40]:
tesla_data.head()

Date        Open        High         Low       Close  \
0 2025-01-06 00:00:00-05:00  423.200012  426.429993  401.700012  411.049988   
1 2025-01-07 00:00:00-05:00  405.829987  414.329987  390.000000  394.359985   
2 2025-01-08 00:00:00-05:00  392.950012  402.500000  387.399994  394.940002   
3 2025-01-10 00:00:00-05:00  391.399994  399.279999  377.290009  394.739990   
4 2025-01-13 00:00:00-05:00  383.209991  403.790009  380.070007  403.309998   

     Volume  Dividends  Stock Splits  
0  85516500        0.0           0.0  
1  75699500        0.0           0.0  
2  73038800        0.0           0.0  
3  62287300        0.0           0.0  
4  67580500        0.0           0.0

## Question 2: Use Webscraping to Extract Tesla Revenue Data


Use the `requests` library to download the webpage https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/revenue.htm Save the text of the response as a variable named `html_data`.


In [41]:
url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/revenue.htm"

In [42]:
response = requests.get(url)

Parse the html data using `beautiful_soup` using parser i.e `html5lib` or `html.parser`.


In [112]:
soup = BeautifulSoup(response.content, 'html.parser')

Using `BeautifulSoup` or the `read_html` function extract the table with `Tesla Revenue` and store it into a dataframe named `tesla_revenue`. The dataframe should have columns `Date` and `Revenue`.


<details><summary>Step-by-step instructions</summary>

```

Here are the step-by-step instructions:

1. Create an Empty DataFrame
2. Find the Relevant Table
3. Check for the Tesla Quarterly Revenue Table
4. Iterate Through Rows in the Table Body
5. Extract Data from Columns
6. Append Data to the DataFrame

```
</details>


<details><summary>Click here if you need help locating the table</summary>

```
    
Below is the code to isolate the table, you will now need to loop through the rows and columns like in the previous lab
    
soup.find_all("tbody")[1]
    
If you want to use the read_html function the table is located at index 1

We are focusing on quarterly revenue in the lab.
```

</details>


In [113]:
tesla_revenue = pd.read_html(str(soup))

Execute the following line to remove the comma and dollar sign from the `Revenue` column. 


In [131]:
tesla_revenue[1]["Tesla Quarterly Revenue (Millions of US $).1"]

0     $21,454
1     $16,934
2     $18,756
3     $17,719
4     $13,757
5     $11,958
6     $10,389
7     $10,744
8      $8,771
9      $6,036
10     $5,985
11     $7,384
12     $6,303
13     $6,350
14     $4,541
15     $7,226
16     $6,824
17     $4,002
18     $3,409
19     $3,288
20     $2,985
21     $2,790
22     $2,696
23     $2,285
24     $2,298
25     $1,270
26     $1,147
27     $1,214
28       $937
29       $955
30       $940
31       $957
32       $852
33       $769
34       $621
35       $615
36       $431
37       $405
38       $562
39       $306
40        $50
41        $27
42        $30
43        $39
44        $58
45        $58
46        $49
47        $36
48        $31
49        $28
50        $21
51        NaN
52        $46
53        $27
Name: Tesla Quarterly Revenue (Millions of US $).1, dtype: object

In [132]:
tesla_revenue[1]["Tesla Quarterly Revenue (Millions of US $).1"] = tesla_revenue[1]["Tesla Quarterly Revenue (Millions of US $).1"].str.replace(',|\$',"")

Execute the following lines to remove an null or empty strings in the Revenue column.


In [133]:
tesla_revenue[1]["Tesla Quarterly Revenue (Millions of US $).1"].dropna(inplace=True)

In [134]:
tesla_revenue = tesla_revenue[1]["Tesla Quarterly Revenue (Millions of US $).1"][tesla_revenue[1]["Tesla Quarterly Revenue (Millions of US $).1"] != ""]

Display the last 5 row of the `tesla_revenue` dataframe using the `tail` function. Take a screenshot of the results.


In [135]:
len(tesla_revenue)

54

## Question 3: Use yfinance to Extract Stock Data


Using the `Ticker` function enter the ticker symbol of the stock we want to extract data on to create a ticker object. The stock is GameStop and its ticker symbol is `GME`.


In [136]:
yf.download('GME', start=start_date, end=end_date)

[*********************100%%**********************]  1 of 1 completed


Open       High        Low      Close  Adj Close   Volume
Date                                                                      
2023-02-17  21.270000  22.070000  21.150000  21.990000  21.990000  2851000
2023-02-21  21.330000  21.830000  20.629999  20.660000  20.660000  3100300
2023-02-22  20.530001  21.190001  20.330000  20.520000  20.520000  2164900
2023-02-23  20.660000  20.870001  19.530001  20.139999  20.139999  2862000
2023-02-24  19.709999  20.080000  19.540001  19.540001  19.540001  2277600
...               ...        ...        ...        ...        ...      ...
2025-01-30  27.840000  28.230000  27.709999  27.990000  27.990000  3343900
2025-01-31  27.790001  28.180000  26.900000  26.900000  26.900000  4544000
2025-02-03  25.570000  26.540001  25.500000  25.889999  25.889999  5740500
2025-02-04  25.850000  26.250000  25.799999  25.900000  25.900000  3353700
2025-02-05  25.700001  25.809999  24.900000  24.930000  24.930000  4531800

[493 rows x 6 columns]

In [137]:
gme = yf.Ticker('GME')

Using the ticker object and the function `history` extract stock information and save it in a dataframe named `gme_data`. Set the `period` parameter to ` "max" ` so we get information for the maximum amount of time.


In [138]:
gme.history(period="max")

Open       High        Low      Close  \
Date                                                                    
2002-02-13 00:00:00-05:00   1.620129   1.693350   1.603296   1.691667   
2002-02-14 00:00:00-05:00   1.712707   1.716074   1.670626   1.683251   
2002-02-15 00:00:00-05:00   1.683250   1.687458   1.658001   1.674834   
2002-02-19 00:00:00-05:00   1.666418   1.666418   1.578047   1.607504   
2002-02-20 00:00:00-05:00   1.615920   1.662210   1.603296   1.662210   
...                              ...        ...        ...        ...   
2025-01-30 00:00:00-05:00  27.840000  28.230000  27.709999  27.990000   
2025-01-31 00:00:00-05:00  27.790001  28.180000  26.900000  26.900000   
2025-02-03 00:00:00-05:00  25.570000  26.540001  25.500000  25.889999   
2025-02-04 00:00:00-05:00  25.850000  26.250000  25.799999  25.900000   
2025-02-05 00:00:00-05:00  25.700001  25.809999  24.900000  24.930000   

                             Volume  Dividends  Stock Splits  
Date                                                          
2002-02-13 00:00:00-05:00  76216000        0.0           0.0  
2002-02-14 00:00:00-05:00  11021600        0.0           0.0  
2002-02-15 00:00:00-05:00   8389600        0.0           0.0  
2002-02-19 00:00:00-05:00   7410400        0.0           0.0  
2002-02-20 00:00:00-05:00   6892800        0.0           0.0  
...                             ...        ...           ...  
2025-01-30 00:00:00-05:00   3343900        0.0           0.0  
2025-01-31 00:00:00-05:00   4544000        0.0           0.0  
2025-02-03 00:00:00-05:00   5740500        0.0           0.0  
2025-02-04 00:00:00-05:00   3353700        0.0           0.0  
2025-02-05 00:00:00-05:00   4531800        0.0           0.0  

[5783 rows x 7 columns]

**Reset the index** using the `reset_index(inplace=True)` function on the gme_data DataFrame and display the first five rows of the `gme_data` dataframe using the `head` function. Take a screenshot of the results and code from the beginning of Question 3 to the results below.


In [139]:
gme_data = pd.DataFrame(gme.info)

In [140]:
gme_data.reset_index(inplace=True)

In [141]:
gme_data.head()

index              address1       city state    zip        country  \
0      0  625 Westport Parkway  Grapevine    TX  76051  United States   
1      1  625 Westport Parkway  Grapevine    TX  76051  United States   
2      2  625 Westport Parkway  Grapevine    TX  76051  United States   

          phone                   website          industry       industryKey  \
0  817 424 2000  https://www.gamestop.com  Specialty Retail  specialty-retail   
1  817 424 2000  https://www.gamestop.com  Specialty Retail  specialty-retail   
2  817 424 2000  https://www.gamestop.com  Specialty Retail  specialty-retail   

   ... returnOnEquity grossProfits freeCashflow operatingCashflow  \
0  ...        0.01504   1194300032    -93387504         -33100000   
1  ...        0.01504   1194300032    -93387504         -33100000   
2  ...        0.01504   1194300032    -93387504         -33100000   

  revenueGrowth  grossMargins ebitdaMargins  operatingMargins  \
0        -0.314       0.26237       0.01037          -0.03558   
1        -0.314       0.26237       0.01037          -0.03558   
2        -0.314       0.26237       0.01037          -0.03558   

   financialCurrency  trailingPegRatio  
0                USD              None  
1                USD              None  
2                USD              None  

[3 rows x 125 columns]

## Question 4: Use Webscraping to Extract GME Revenue Data


Use the `requests` library to download the webpage https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/stock.html. Save the text of the response as a variable named `html_data_2`.


In [142]:
url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/stock.html"

In [143]:
response = requests.get(url)

Parse the html data using `beautiful_soup` using parser i.e `html5lib` or `html.parser`.


In [144]:
soup = BeautifulSoup(response.content, 'html.parser')

Using `BeautifulSoup` or the `read_html` function extract the table with `GameStop Revenue` and store it into a dataframe named `gme_revenue`. The dataframe should have columns `Date` and `Revenue`. Make sure the comma and dollar sign is removed from the `Revenue` column.


> **Note: Use the method similar to what you did in question 2.**  


<details><summary>Click here if you need help locating the table</summary>

```
    
Below is the code to isolate the table, you will now need to loop through the rows and columns like in the previous lab
    
soup.find_all("tbody")[1]
    
If you want to use the read_html function the table is located at index 1


```

</details>


In [145]:
gme_revenue = pd.read_html(str(soup))

In [153]:
gme_revenue = gme_revenue[1]["GameStop Quarterly Revenue (Millions of US $).1"]

Display the last five rows of the `gme_revenue` dataframe using the `tail` function. Take a screenshot of the results.


In [154]:
gme_revenue.tail()

57    $1,667
58      $534
59      $416
60      $475
61      $709
Name: GameStop Quarterly Revenue (Millions of US $).1, dtype: object

## Question 5: Plot Tesla Stock Graph


Use the `make_graph` function to graph the Tesla Stock Data, also provide a title for the graph. Note the graph will only show data upto June 2021.


<details><summary>Hint</summary>

```

You just need to invoke the make_graph function with the required parameter to print the graphs.The structure to call the `make_graph` function is `make_graph(tesla_data, tesla_revenue, 'Tesla')`.

```
    
</details>


In [162]:
tesla_data.columns

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Volume', 'Dividends',
       'Stock Splits'],
      dtype='object')

In [163]:
make_graph(tesla_data,tesla_revenue,"tsla")

AttributeError: 'Series' object has no attribute 'Date'

## Question 6: Plot GameStop Stock Graph


Use the `make_graph` function to graph the GameStop Stock Data, also provide a title for the graph. The structure to call the `make_graph` function is `make_graph(gme_data, gme_revenue, 'GameStop')`. Note the graph will only show data upto June 2021.


<details><summary>Hint</summary>

```

You just need to invoke the make_graph function with the required parameter to print the graphs.The structure to call the `make_graph` function is `make_graph(gme_data, gme_revenue, 'GameStop')`

```
    
</details>


<h2>About the Authors:</h2> 

<a href="https://www.linkedin.com/in/joseph-s-50398b136/">Joseph Santarcangelo</a> has a PhD in Electrical Engineering, his research focused on using machine learning, signal processing, and computer vision to determine how videos impact human cognition. Joseph has been working for IBM since he completed his PhD.

Azim Hirjani


## Change Log

| Date (YYYY-MM-DD) | Version | Changed By    | Change Description        |
| ----------------- | ------- | ------------- | ------------------------- |
| 2022-02-28        | 1.2     | Lakshmi Holla | Changed the URL of GameStop |
| 2020-11-10        | 1.1     | Malika Singla | Deleted the Optional part |
| 2020-08-27        | 1.0     | Malika Singla | Added lab to GitLab       |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>

<p>
